iwyu.cc

```diff
@@ -2434,6 +2424,26 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     return true;
   }
 
+  bool VisitDeclRefExpr(clang::DeclRefExpr* expr) {
+    if (CanIgnoreCurrentASTNode())
+      return true;
+    // Report EnumName instead of EnumName::Item.
+    // It supports for removing EnumName forward- (opaque-) declarations
+    // from output.
+    if (const auto* enum_constant_decl =
+            dyn_cast<EnumConstantDecl>(expr->getDecl())) {
+      const auto* enum_decl =
+          cast<EnumDecl>(enum_constant_decl->getDeclContext());
+
+      // for unnamed enums, enumerator name is still preferred
+      if (enum_decl->getIdentifier())
+        ReportDeclUse(CurrentLoc(), enum_decl);
+      else
+        ReportDeclUse(CurrentLoc(), enum_constant_decl);
+    }
+    return Base::VisitDeclRefExpr(expr);
```

Comment:
I just realized that these `Base::` calls are relatively recent mistakes (of mine). They are not necessary in `IwyuBaseAstVisitor`. See https://github.com/include-what-you-use/include-what-you-use/pull/1086.